src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -426,41 +478,47 @@ def get_vm_set_type(self, enable_validation: bool = False, **kwargs):
         read_from_mc = False
         if value_obtained_from_mc is not None:
             vm_set_type = value_obtained_from_mc
-            # clean up intermediate if `mc` has been decorated
-            self.remove_intermediate(parameter_name)
             read_from_mc = True
-        elif intermediate is not None:
-            vm_set_type = intermediate
         else:
             vm_set_type = raw_value
 
-        dynamic_completion = False
-        # check whether the parameter meet the conditions of dynamic completion
-        if not vm_set_type:
-            dynamic_completion = True
+        # skip dynamic completion & validation if option read_only is specified
+        if kwargs.get("read_only"):
+            return vm_set_type
+
+        # the value verified by the validator may have case problems, and the
+        # "_set_vm_set_type" function will adjust it
+        dynamic_completion = True
```

Comment:
always true?

Reply:
The server side is sensitive to the case of the value, but the user may enter a value with the wrong case, which can be corrected here. The reason why `dynamic_completion ` is set to True by default is to be consistent with the behavior of the code in cli.

In addition, if the value is obtained from the mc object, no further modifications will be made (the next line of code will change `dynamic_completion`  to False in this case).